docs/source/reference/api-server/api-server-tunning.rst

```diff
@@ -0,0 +1,144 @@
+
+        .. code-block:: diff
+
+            resources:
+            -   cpus: 8+
+            -   memory: 16+
+            +   cpus: 16+
+            +   memory: 32+
+
+The following table shows the maximum concurrency for different resource configurations:
+
+.. list-table::
+   :widths: 1 1 2 2
+   :header-rows: 1
+
+   * - CPU
+     - Memory
+     - Long requests
+     - Short requests
+   * - 4
+     - 8Gi
+     - 8 Max concurrency
+     - 11 Max concurrency
+   * - 16
+     - 32Gi
+     - 32 Max concurrency
+     - 60 Max concurrency
+   * - 32
+     - 64Gi
+     - 64 Max concurrency
+     - 145 Max concurrency
+   * - 64
+     - 128Gi
+     - 128 Max concurrency
+     - 299 Max concurrency
+   * - 128
+     - 256Gi
+     - 256 Max concurrency
+     - 589 Max concurrency
+
+Queuing requests and polling status asynchronously
+--------------------------------------------------
+
+There is no limit on the number of queued requests. So in addition to increasing the allocated resources to improve the maximum concurrency, you can also submit requests with ``--async`` flag and poll the status asynchronously to avoid blocking. For example:
+
+.. code-block:: bash
+
+    # Submit 2000 jobs at once without blocking
+    for i in `seq 1 2000`; do
+        sky jobs launch --async job.yaml
+    done
+    # Poll the status of the jobs
+    sky jobs queue
+
+The requests will be queued on the server and be processed in submission order. If you find the status is not updated for a while, you can inspect the status of the submitted requests with:
+
+.. code-block:: console
+
+    $ sky api status
+    ID                                    User             Name             Created         Status
+    d9c95c7e-d248-4a7f-b72e-636511405357  alice            sky.jobs.launch  a few secs ago  PENDING
+    767182fd-0202-4ae5-b2d7-ddfabea5c821  alice            sky.jobs.launch  a few secs ago  PENDING
+    5667cff2-e953-4b80-9e5f-546cea83dc59  alice            sky.jobs.launch  a few secs ago  RUNNING
+
+There should be some ``RUNNING`` requests that occupy the concurrency limit. Usually the ``RUNNING`` requests make progress and finally your requests will be processed, but if the ``RUNNING`` requests are stuck, you can inspect the request log with:
+
+.. code-block:: console
+
+    # Replace <request_id> with the actual request id from the ID column
+    $ sky api logs <request_id>
+
+If the request is stuck according to the log, e.g. retrying to launch VMs that is out of stock, you can cancel the request with:
+
+.. code-block:: bash
+
+    sky api cancel <requst_id>
+
+Avoid concurrent logs requests
+------------------------------
+
+Though a job can run for any length of time, concurrent jobs does not occupy the concurrency. Because once a job is submitted to the cluster, it will be detached and no longer takes any resources off the API server.
```

Comment:
Should we emphasize this earlier in this page to assure people that the `long requests` does not count the running jobs.